src/components/Popup/Popup.tsx

```diff
@@ -224,10 +256,54 @@ export default class Popup extends AutoControlledComponent<Extendable<PopupProps
     )
   }
 
+  getTriggerProps(triggerElement) {
+    const triggerProps: any = {}
+
+    const { on } = this.props
+
+    if (on === 'click') {
+      triggerProps.onClick = (e, ...rest) => {
+        this.trySetOpen(!this.state.open, e)
+        _.invoke(triggerElement, 'props.onClick', e, ...rest)
+      }
+    }
+    if (on === 'hover') {
```

Comment:
this might be for a separate PR, but if popup appears on hover, it also needs to appear on click for the case when user uses keyboard or virtual cursor to access the popup

Reply:
On `click` or `focus`? And how can I close that popup?
FWIW, Semantic displays popup on space or Enter with `on='hover'`, it is possible to close it with Esc only (no Enter/space).
With `on='click'` Enter/space can both open and close the popup.
It also supports `on='focus'` (https://react.semantic-ui.com/modules/popup/#usage-focus) and array for `on` prop.

Reply:
Couple of thoughts from my side:
I started with the ideas from Semantic UI, but there are problems when for some popup we have the `hover` and `click` **at the same time** (awkward things are happening because we have both click and hover opening/closing the popup because, so if you go on the popup to click it in order for it to be opened, the hovered is already opening it etc - please try this on SUIR)
I cannot imagine popup that would need to implement the click and hover at the same time (maybe I am wrong with this assumption)
The focus on the other hand, should follow the (click or hover triggering) - I would image that popup with hover would also be opened if we focus it and popup with click event would be opened if we focus on it and press enter/space. 
@miroslavstastny what do you think about this points? @jurokapsiar can you reference some best practices about how should this work?

Reply:
So, in conclusion are we okay with following the SUIR practices? (Still not sure tough for the hover + click at the same time)

Reply:
@layershifter is ok with excluding the combination of having click and hover at the same time. Will proceed with this, and the rest will follow the implementation in suir